### DISCUSSION TITLE
Improving the VoiceOver experience with more succinct labels + custom content

### DISCUSSION THREAD

**Discussion by BasThomas** [community/community#13922] [Mobile] [answered]
Hi all!

I love that the GitHub (iOS, I don't have experience with the Android one) app is so accessible. It's very clear that you've put a lot of thought and effort into accessibility!

One thing I've noticed, however, is that sometimes, accessibility labels can become quite long, including lots of data. That can make it difficult for a VoiceOver user to efficiently navigate the screens.

What we do to make accessibility labels shorter, while not "throwing out" any important data, is [Customized Accessibility content](https://developer.apple.com/documentation/accessibility/customized_accessibility_content) / [Custom content](https://www.basbroek.nl/improving-voiceover#custom-content).

Let me show you a video of an example of a long label in the GitHub app, as well as an example where `AXCustomContent` is used to make VoiceOver navigation more efficient.

## GitHub

https://user-images.githubusercontent.com/4190298/160913186-e8892023-6af0-4f20-a415-f2ed2fa6287d.mov

## Example usage of `AXCustomContent`

https://user-images.githubusercontent.com/4190298/160913098-7cc60aa8-e29e-4e86-bb56-e583f9d1cabc.mp4

**Comment by HenSquared** [comment, marked as answer]
@BasThomas, we really appreciate the time you spent writing up and recording this! Your recordings explaining the issue and an example solution are really helpful.

I created an internal issue to track this and hope we can improve this experience soon. Thanks!

**Reply by BasThomas** [reply]
Awesome; looking forward! Let me know if there's anything I can do to help. :)
